erigon-lib/state/domain_shared.go

```diff
@@ -112,14 +113,21 @@ type HasAgg interface {
 	Agg() any
 }
 
-func NewSharedDomains(tx kv.Tx, logger log.Logger) (*SharedDomains, error) {
+func NewSharedDomains(db kv.RoDB, logger log.Logger) (*SharedDomains, error) {
 
 	sd := &SharedDomains{
 		logger:  logger,
 		storage: btree2.NewMap[string, dataWithPrevStep](128),
 		//trace:   true,
 	}
-	sd.SetTx(tx)
+
+	tx, err := db.BeginRo(context.Background())
```

Comment:
I think erigon has 1 problem. It does: 
```
tx:=beginrw
unwind(tx)
runForward(tx)
tx.commit()
```
and such creation of new rotx - of course will make invisible all made by `unwind` writes. 

but we may change Erigon - and commit after unwind. it will make us a bit slower on chain-tip, but I also have solution for it: use of `SafeNoSync` for chaindata. Elton does prototyping this feature for `nodesDB` now: https://github.com/erigontech/erigon/issues/13550 - it will make `rwtx.Commit` cheap (because fsync will happen in background).